### ISSUE TITLE
Resolve SettingWithCopyWarning When Using LabelEncoder in Pandas Datafram

### ISSUE THREAD

**Issue by karsterr** [karsterr/DevSalary#1]

**Comment by karsterr** [comment]
I'm encountering a SettingWithCopyWarning while trying to use LabelEncoder from scikit-learn to encode categorical variables in my pandas DataFrame. Here's a simplified version of my code:

``python
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Load the dataset
df = pd.read_csv("cleaned_data.csv").copy()

# Initialize LabelEncoder
label_encoder = LabelEncoder()

# Encode 'position' and 'level' columns using .loc to avoid SettingWithCopyWarning
df.loc[:, 'position'] = label_encoder.fit_transform(df['position'])
df.loc[:, 'level'] = label_encoder.fit_transform(df['level'])

# Separate independent variables (X) and dependent variable (y)
X = df.drop(columns=['timestamp', 'salary_currency', 'salary'])
y = df['salary']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Create and train the linear regression model
model = LinearRegression()
model.fit(X_train, y_train)

# Make predictions on the training dataset
train_predictions = model.predict(X_train)

# Make predictions on the test dataset
test_predictions = model.predict(X_test)

# Evaluate the performance of the model
train_rmse = mean_squared_error(y_train, train_predictions, squared=False)
test_rmse = mean_squared_error(y_test, test_predictions, squared=False)

print("Training set RMSE:", train_rmse)
print("Test set RMSE:", test_rmse)
```

How can I resolve this SettingWithCopyWarning and encode my categorical variables correctly? Any help or suggestions would be greatly appreciated!

The issue arises when I try to use .fit_transform() from LabelEncoder to encode the 'position' and 'level' columns. I've read the documentation and tried using .loc as suggested, but the warning persists.
